cobalt/browser/cobalt_web_contents_observer.cc

```diff
@@ -53,9 +61,20 @@ void CobaltWebContentsObserver::RegisterInjectedJavaScript() {
   }
 }
 
-// Placeholder for a WebContentsObserver override
 void CobaltWebContentsObserver::PrimaryMainDocumentElementAvailable() {
-  LOG(INFO) << "Cobalt::PrimaryMainDocumentElementAvailable";
+  content::RenderFrameHost* render_frame_host =
+      web_contents()->GetPrimaryMainFrame();
+  if (!migration_attempted.load()) {
```

Comment:
Prefer early return:
```
if (migration_attempted.load()) {
  return;
}
```

I'm a bit unconvinced of |migration_attempted| being a file-static variable. Since CobaltWebContentsObserver is a class, I feel that this should be a static member. But more importantly, I think it's not CobaltWebContentsObserver's business to know whether the migration has been attempted before or not: Ideally, the files migrate.(cc|h) would contain a class (e.g. CookieMigrationManager or CookieMigrationController or some such) and from here you would just call its Clear/Migrate methods; inside of those methods, the class would have a static member to mark whether such migration had been attempted before or not.